sound/soc/sof/ipc4-topology.h

```diff
@@ -357,4 +357,35 @@ struct sof_ipc4_process {
 	bool payload_with_output_fmt;
 };
 
+/**
+ * struct sof_ipc4_pin_format - widgets pin format
+ * @pin_index: pin index for the widget
```

Comment:
index relative to what? why is the benefit of having an index if there's no information on whether the pin is used for input or output.

Reply:
The structure could be used for input/output pin, if it is for input pin, pin_index is the input pin index, e.g. if a widget has three input pins, we could have an array with three elements of this structure, with pin_index=0,1,2 separately.

The structure is used to describe pin format for a single pin, instead of having a flag to indicate input/output, it depends on how user uses this structure.  the common use case is like what we use it in struct sof_ipc4_base_module_cfg_ext:
a single pin format array, with size == (num_sink_pin_format + num_source_pin_formats). 
Elements of `Index=0 ~ num_sink_pin_format` are for sink pins, and `index=num_sink_pin_format ~ num_source_pin_formats` are for source pins.

Reply:
then add a comment that "the pin can be used for source or sink cases, and the index is relative to the type of pin being used".